contribs/src/main/scala/io/delta/storage/BaseExternalLogStore.scala

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.storage
+
+import java.io.FileNotFoundException
+import java.net.URI
+import java.nio.charset.StandardCharsets.UTF_8
+import java.nio.file.FileSystemException
+
+import com.google.common.io.CountingOutputStream
+import org.apache.commons.io.IOUtils
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs._
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.delta.metering.DeltaLogging
+import org.apache.spark.sql.delta.util.FileNames
+import org.apache.spark.sql.delta.storage.HadoopFileSystemLogStore
+
+abstract class BaseExternalLogStore(
```

Comment:
class doc. Why does this abstract class exist? (used so that we can use an external store child implementation to provide the mutual exclusion that the cloud store, e.g. s3, is missing)